docs/node-groups.md

```diff
@@ -141,6 +141,40 @@ Check the following references as you may desire:
 *  [AWS NVMe drivers for Windows instances](https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/aws-nvme-drivers.html)
 *  [EC2 Instance Update – M5 Instances with Local NVMe Storage (M5d)](https://aws.amazon.com/blogs/aws/ec2-instance-update-m5-instances-with-local-nvme-storage-m5d/)
 
+### Block Device Encryption
+
+Your worker nodes are always going to have encrypted block devices.
+
+#### Default Encryption with KMS AWS Managed Key
+
+By default it uses `aws/ebs` KMS AWS Managed Key in case you assume the default values for `worker_create_kms_key` and `worker_kms_key_arn` variables.
+
+#### Encryption with created KMS Customer Managed Key
+
+You are able to use a KMS Customer Managed Key, created during infrastructure provisioning process, by simplying setting `worker_create_kms_key` variable to `true`.
+
+```hcl
+  ...
+    worker_create_kms_key = true
```

Comment:
this variable reads to me like it is not a customer managed key. Should we changes to `customer_managed_kms_key` = true? Or am I misreading the doc?

Reply:
There are two different variables `worker_create_kms_key` which default value is `false`, and `worker_kms_key_arn` which default value is `null`.

The idea of using two different variables is to cover three different scenarios:

1. When keeping these variable with the default values, the self managed nodes are encrypted with default **aws/ebs** AWS Managed Key.
2. When setting `worker_create_kms_key` to `true`, the framework creates a KMS Customer Managed Key which is used to encrypt the ebs volumes.
3. When setting `worker_kms_key_arn` to a valid kms arn, the framework uses the informed key to encrypt the ebs volumes.

If we change the variable `worker_create_kms_key` to `customer_managed_kms_key`, isn't it going to make a confusion for customers with the `worker_kms_key_arn` ? Do you suggest covering only scenarios one and two from above so we can accept only kms key ?